### PR TITLE
Add node placement for proxy server

### PR DESCRIPTION
**Pull request by qiujian16**
also enable the test

Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

File: pkg/proxyserver/controllers/clustermanagementaddon_controller.go [hunk 1]
```diff
@@ -90,13 +89,10 @@ type ClusterManagementAddonReconciler struct {
...
 
 	newCertRotatorFunc func(namespace, name string, sans ...string) selfsigned.CertRotation
-	proxyConfigClient  proxyclient.Interface
-	proxyConfigLister  proxylister.ManagedProxyConfigurationLister
-	addonLister        addonlister.ManagedClusterAddOnNamespaceLister
-	clusterAddonLister addonlister.ClusterManagementAddOnLister
 }
 
 func (c *ClusterManagementAddonReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// TODO should add a filter to only watch addon with cluster-proxy name
```
Comment (yue9944882) [inline pkg/proxyserver/controllers/clustermanagementaddon_controller.go:95]: will fix this in a follow-up

File: pkg/proxyserver/controllers/suite_test.go [hunk 2]
```diff
@@ -53,26 +75,60 @@ var _ = BeforeSuite(func() {
...
+	selfSigner, err := selfsigned.NewSelfSignerFromSecretOrGenerate(kubeClient, "default", "test-ca")
+	Expect(err).NotTo(HaveOccurred())
 
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	err = RegisterClusterManagementAddonReconciler(
+		mgr, selfSigner, kubeClient, kubeInformer.Core().V1().Secrets())
 	Expect(err).NotTo(HaveOccurred())
-	Expect(k8sClient).NotTo(BeNil())
 
+	fmt.Printf("start manager")
```
Comment (yue9944882) [inline pkg/proxyserver/controllers/suite_test.go:119]: nit: use `By(...)` instead of `fmt.Print`?

File: pkg/proxyserver/controllers/manifests.go [hunk 3]
```diff
@@ -26,6 +26,10 @@ func newOwnerReference(config *proxyv1alpha1.ManagedProxyConfiguration) metav1.O
 
 func newServiceAccount(config *proxyv1alpha1.ManagedProxyConfiguration) *corev1.ServiceAccount {
 	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
```
Comment (yue9944882) [inline pkg/proxyserver/controllers/manifests.go:29]: they're applied to the hub cluster using direct create/update api calls, only those applied by the manifest-work will need the explicit type meta, is it?

**Review by yue9944882 — COMMENTED** [review]
the integration test looks great, a few nits then lets get in

File: pkg/proxyserver/controllers/manifests.go (hunk 3 above)
Comment (qiujian16) [reply pkg/proxyserver/controllers/manifests.go]: yeah, removed

File: pkg/proxyserver/controllers/clustermanagementaddon_controller.go [hunk 4]
```diff
@@ -50,6 +47,8 @@ var _ reconcile.Reconciler = &ClusterManagementAddonReconciler{}
 
 var log = ctrl.Log.WithName("ClusterManagementAddonReconciler")
 
+const addonName = "cluster-proxy"
```
[the remark this answers is not in the document]
Comment (qiujian16) [reply pkg/proxyserver/controllers/clustermanagementaddon_controller.go]: done

**Comment by xuezhaojun** [comment]
Do we need to update CRD in chart as well? https://github.com/open-cluster-management-io/cluster-proxy/blob/main/charts/cluster-proxy/crds/managedproxyconfigurations.yaml

**Comment by xuezhaojun** [comment]
Do we need squash two commits into one?

**Comment by qiujian16** [comment]
helm chart updated, I think prow will auto squash the commit..

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go [hunk 5]
```diff
@@ -189,25 +203,28 @@ type ManagedProxyConfigurationProxyServer struct {
 	// Note that the replicas will also be reflected in the flag `--server-count`
 	// so that agents can discover all the server instances.
 	// +optional
-	// +kubebuilder:default=3
```
Comment (yue9944882) [inline pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go:192]: @qiujian16 i missed this line, is it intended?

Comment (qiujian16) [reply pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go:192]: probably not...

**Pull request merged**